### PR TITLE
feat: move range filter to list pages header

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Moves the range filter from the top paging component to the header of the lists page.

It will only appear if you pass the `ListPageLayout` a `setRange` prop. As well the `TableHeader` component can still include a range filter, but it will only appear if it receives a `setRangeFilter` and `range` prop.

To see the changes on the List Contributors page, visit e.g. https://deploy-preview-1817--oss-insights.netlify.app/lists/23c2ca69-7667-4b6a-ad2e-d402999af310/contributors

Relates to the changes proposed for the range filter in #1815.

**Before**

CleanShot 2023-10-06 at 06 10 59

**After**

CleanShot 2023-10-06 at 06 09 36

You can see the range filter in action on the contributors page

https://github.com/open-sauced/app/assets/833231/37246229-d0c4-4deb-a891-a3c8f9f3a662

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #1815 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx [hunk 1]
```diff
@@ -7,9 +7,14 @@ import Checkbox from "components/atoms/Checkbox/checkbox";
...
   selected?: boolean;
   handleOnSelectAllContributor?: (contributor: any) => void;
+  range?: number;
 }
 
-const ContributorListTableHeaders = ({ selected, handleOnSelectAllContributor }: ContributorListTableHeadersProps) => {
+const ContributorListTableHeaders = ({
+  selected,
+  handleOnSelectAllContributor,
+  range = 30,
```
Comment (nickytonline) [inline components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx:16]: Range is passed in now so when you filter, the graph associated with that column can be updated to reflect the correct range filter.

File: components/molecules/ContributorListTableRow/contributor-list-table-row.tsx [hunk 2]
```diff
@@ -47,14 +48,15 @@ const ContributorListTableRow = ({
   topic,
   selected,
   handleOnSelectContributor,
+  range,
 }: ContributorListTableRow) => {
   const [tableOpen, setTableOpen] = useState(false);
 
   const { data: user } = useFetchUser(contributor.author_login);
-  const { data } = useContributorPullRequests(contributor.author_login, topic, [], 30);
+  const { data } = useContributorPullRequests(contributor.author_login, topic, [], range);
```
Comment (nickytonline) [inline components/molecules/ContributorListTableRow/contributor-list-table-row.tsx:56]: For the list contributors page, I discovered that I wasn't passing the range filter

File: components/molecules/TableHeader/table-header.tsx [hunk 3]
```diff
@@ -95,12 +95,7 @@ const TableHeader = ({
         />
       </div>
       <div className="flex flex-col-reverse items-start gap-3 md:flex-row md:items-end ">
-        {range ? (
-          <ComponentDateFilter setRangeFilter={(range: number) => setRangeFilter?.(range)} defaultRange={range} />
-        ) : (
-          ""
-        )}
-
+        {setRangeFilter && range ? <ComponentDateFilter setRangeFilter={setRangeFilter} defaultRange={range} /> : null}
```
Comment (nickytonline) [inline components/molecules/TableHeader/table-header.tsx:98]: Only display the range filter in the header now if the `setRangeFilter` and `range` props are passed in.

File: components/organisms/ContributorsTable/contributors-table.tsx [hunk 4]
```diff
@@ -8,6 +8,7 @@ export interface ContributorTableProps {
   loading?: boolean;
   selectedContributors?: DbPRContributor[];
   handleSelectContributors?: (state: boolean, contributor: DbPRContributor) => void;
+  range?: number;
```
Comment (nickytonline) [inline components/organisms/ContributorsTable/contributors-table.tsx:11]: Marked as optional and defaults to 30 days so existing stuff doesn't break.

**Review by brandonroberts — APPROVED** [review]

**Comment by isabensusan** [comment]
Hey! Can we align the date filter to the red line? So it's aligned with the table and the "Share" button. (and while you're at it, if you could align the "Showing:10 per page" dropdown to the red line that would be **ideal** too)

image

**Comment by isabensusan** [comment]
and thanks for the great work btw @nickytonline !! so fast ⚡️🏎️

**Comment by nickytonline** [comment]
@isabensusan, it looks like the Share button is positioned absolute, so in the screenshot I send you it looked good, but as I resized the screen it doesn't adjust. Can you create a separate issue for that. In the meantime, I have the grid header aligning properly, just sorting out the range filter.

CleanShot 2023-10-06 at 11 19 17

**Review by OgDev-01 — APPROVED** [review]

**Review by isabensusan — APPROVED** [review]
Thanks for looking into the aligning issue! I think this PR is ready to close even if we couldn't get that 100% fixed. I'll go ahead an open an issue to improve component alignment separately

**Pull request merged**